### DISCUSSION TITLE
Strange behavior with templ-vscode

### DISCUSSION THREAD

**Discussion by danawoodman** [a-h/templ#225] [Q&A]
I'm noticing some weird formatting changes happening with the templ-vscode extension and "Format on Save" enabled.

If I attempt to remove the leading/trailing spaces on the brackets then save I get a compounding formatting artifact

![Screen Recording 2023-10-09 at 04 48 03 PM](https://github.com/a-h/templ/assets/157695/0d2eadda-573f-4c11-b064-16183b3068bf)

Changing to plain text does not have this issue, so I'm assuming it is templ-vscode

```
go 1.21.0
github.com/a-h/templ v0.2.364
```

**Comment by a-h** [comment]
Thanks for reporting the issue! I've heard another report of this behaviour at https://github.com/a-h/templ/issues/192, but haven't yet been able to reproduce it.

Is this happening on Windows?

I'm wondering whether it's a UTF-16 vs UTF-8 bug in the VS Code plugin, or the templ LSP.

**Reply by danawoodman** [reply]
It's on macOS. I just upgraded to Sonoma and now getting  a whole set of other errors that prevent me from testing this issue again. Not sure what changed to cause this

![image](https://github.com/a-h/templ/assets/157695/35f51bee-df3e-458e-be87-261b2c455301)

`templ` is on my path:

![image](https://github.com/a-h/templ/assets/157695/38c3d952-5b5f-45db-8e3c-4d48405b2cb8)

EDIT: lol, restarting VSCode fixed that issue (but not the text expanding one)

EDIT 2: I'm not seeing the issue now after removing the bracket code and re-creating it. I will report back if it returns

**Reply by ross96D** [reply]
I have encountered this same formating issue in Ubuntu using vscode
